janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/strategy/JanusGraphLocalQueryOptimizerStrategy.java

```diff
@@ -92,42 +79,27 @@ public void apply(final Traversal.Admin<?, ?> traversal) {
                 vertexStep.setLimit(0, QueryUtil.mergeHighLimits(limit, vertexStep.getHighLimit()));
             }
 
-            if (useMultiQuery) {
-                vertexStep.setUseMultiQuery(true);
-            }
-
-            if (janusGraph.getConfiguration().batchPropertyPrefetching()) {
-                applyBatchPropertyPrefetching(traversal, vertexStep, nextStep, janusGraph.getConfiguration().getTxVertexCacheSize());
+            if (batchPropertyPrefetching) {
+                applyBatchPropertyPrefetching(originalStep.getTraversal(), vertexStep, nextStep, txVertexCacheSize);
             }
         });
+    }
 
-
-        /*
-                ====== PROPERTIES STEP ======
-         */
-
-
-        TraversalHelper.getStepsOfClass(PropertiesStep.class, traversal).forEach(originalStep -> {
+    private void applyJanusGraphPropertiesSteps(Admin<?, ?> traversal) {
+        TraversalHelper.getStepsOfAssignableClass(PropertiesStep.class, traversal).forEach(originalStep -> {
             final JanusGraphPropertiesStep propertiesStep = new JanusGraphPropertiesStep(originalStep);
-            TraversalHelper.replaceStep(originalStep, propertiesStep, traversal);
-
+            TraversalHelper.replaceStep(originalStep, propertiesStep, originalStep.getTraversal());
```

Comment:
Due to the (now fixed) execution order of traversal strategies, I had to experiment a bit with `getStepsOfAssignableClass()` and `getStepsOfAssignableClassRecursively()` in order to find the best solution.
I often found myself in a situation where using `traversal` directly was error-prone because it always references the outer traversal when switching to `getStepsOfAssignableClassRecursively()`. Because we always want the new step to be in the same traversal as the old step, I switched to using `originalStep.getTraversal()` here which is much more robust. 
You are right that this is not explicitly needed anymore but I don't think it hurts and this change has the potential to save hours of debugging.